web-access-control/protected-operation/read-inherited-access-public.feature

```diff
@@ -0,0 +1,137 @@
+Feature: Public agents can read (and only that) a resource when granted inherited read access
+  Background: Create test resources with correct access modes
+    * def authHeaders = (method, url, public) => !public ? clients.bob.getAuthHeaders(method, url) : {}
+    * def createResources =
+    """
+      function (modes) {
+        const testContainer = rootTestContainer.createContainer()
+        if (modes.includes('write')) {
+          testContainer.accessDataset = testContainer.accessDatasetBuilder
+            //.setPublicAccess(testContainer.url, webIds.bob, ['write'])
```

Comment:
Is that an out-commented line? I got a bit confused here.

Reply:
Thanks for pointing this out. I meant to revisit it. I thought this was to do with the write-without-read test and commented the lines out while I decided that I would have that test within the write set, or separate. I should have come back and removed the commented out lines along with the Scenario at the end of this file.

However, now I am thinking again. To make the "cannot `DELETE`" tests valid should I grant the test agent `write` access to the container such that the reason they cannot delete the resource is due to the read only access on it, not the lack of write on the container? This would be a change to all 6 features since it is independent to whether the access is direct or inherited. What do you think?

Reply:
I just have to admit I don't understand the connections :-)

I think we should have tests but for the situation where `DELETE` fails due to that write is missing on the container and when it is missing on the resource itself.

But I don't quite understand how that relates to this feature file... :-)

Reply:
This test focuses on what you can do with read access so granting read to a resource shouldn't let you delete it. However not granting write to the container would be another reason you couldn't delete it so to focus this test on the read only access to the resource, the user should have write access to the container - the only thing stopping them delete the resource is it's read access, nothing else. So that covers the second part of your sentence. 

Then the  write tests would probably be the place where you determine that you need write on the container and the resource to delete it.